src/packet.rs

```diff
@@ -149,7 +149,7 @@ mod tests {
             AprsData::Message(msg) => {
                 assert_eq!(msg.addressee, "DEST");
                 assert_eq!(msg.text, "Hello World! This msg has a : colon ");
-                assert_eq!(msg.id, Some(32975));
+                assert_eq!(msg.id, Some("3a2B975".to_string()));
```

Comment:
Minor: instead of `.to_string()` on the RHS, you could use `msg.id.as_deref()` for the LHS

Reply:
I think `assert_eq!(msg.id, Some("3a2B975"))` would work on it's own. [String implements `PartialEq<str>`](https://doc.rust-lang.org/std/string/struct.String.html#impl-PartialEq%3C%26%27a%20str%3E).

Reply:
Good catch, should be fixed now